mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -777,6 +787,20 @@ class LogisticRegressionModel private[spark] (
     throw new SparkException("No training summary available for this LogisticRegressionModel")
   }
 
+  @Since("2.1.0")
+  def binarySummary: BinaryLogisticRegressionTrainingSummary = summary match {
+    case b: BinaryLogisticRegressionTrainingSummary => b
+    case _ => throw new SparkException("Summary type dismatch, " +
```

Comment:
Also, the current error doesn't really help the user.  How about "Cannot create a binarySummary for a non-binary model (with numClasses = $numClasses).  Use multinomialSummary instead."